lib/rules/camelcase.js

```diff
@@ -71,6 +76,11 @@ module.exports = {
         if (properties !== "always" && properties !== "never") {
             properties = "always";
         }
+        const ignorePattern = options.ignorePattern || null;
+
+        if (ignorePattern) {
+            config.ignorePattern = new RegExp(ignorePattern);
```

Comment:
Nitpick: Instead of using a regex which is sometimes null, can you use a regex that never matches anything as the default? That way, you won't need to do null-checks later.

For example:

```js
const ignorePattern = new RegExp(options.ignorePattern || "^$");
```

Then in `isUnderscored`, you can just use:

```js
return !ignorePattern.test(name) && name.indexOf("_") > -1 && /* ... */
```